### PR TITLE
[explorer] owned coins view changes

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

image

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by mystie711** [comment]
@Nikhil-Mysten i am unsure whether we are adding the coin icons as part of this work or not?
Is this task implementing this Figma design spec?
https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?type=design&node-id=9125-308742&mode=design&t=fbauxpqIq6hifHrr-4

**Comment by Nikhil-Mysten** [comment]
@mystie711 Added coin icons + sorting!

**Comment by mystie711** [comment]
Nice. Thanks @Nikhil-Mysten !

How about the expanded view update with the new designs? Different PR?

**Comment by mystie711** [comment]
Additionally, we don't need to limit the coin names to just four characters now that we have the Unrecognized coins label. Could we expand the characters limit to show atleast 12 characters before ellipsizing?

Also take a closer look at the colors, spacing of text and each row item, border/line colors and such
image

I also notice some delta in the hover over feedback - should be rounded corners.

Please send me an Explorer Vercel link on Slack so i can do a review. I feel like it would be best if we do a quick slack huddle on visual details. Might be efficient.

File: apps/explorer/src/components/OwnedCoins/CoinIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useCoinMetadata } from '@mysten/core';
```
Comment (plam-ml) [inline apps/explorer/src/components/OwnedCoins/CoinIcon.tsx:4]: Can we use this component or expand this? https://github.com/MystenLabs/sui/blob/main/apps/explorer/src/ui/CoinsStack.tsx

File: apps/explorer/src/components/OwnedCoins/ImageIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { cva, type VariantProps } from 'class-variance-authority';
```
Comment (plam-ml) [inline apps/explorer/src/components/OwnedCoins/ImageIcon.tsx:4]: I think we have something similar alrdy here https://github.com/MystenLabs/sui/blob/main/apps/explorer/src/ui/ImageIcon.tsx

File: apps/explorer/src/components/OwnedCoins/CoinIcon.tsx (hunk 1 above)
Comment (Nikhil-Mysten) [reply apps/explorer/src/components/OwnedCoins/CoinIcon.tsx:4]: they've got some decently different styling so I think let's leave this in here

**Review by plam-ml — APPROVED** [review]

**Pull request merged**